### PR TITLE
added KVFilter, added AtomicSwitch on Drain ...

### PR DESCRIPTION
**Pull request by przygienda**
Both very useful on large scale systems. API stayed the same ...

### CODE REVIEW

File: src/file.rs [hunk 1]
```diff
@@ -69,6 +74,19 @@ impl FileLoggerBuilder {
         self
     }
 
+    /// Sets KVFilter
```
Comment (sile) [inline src/file.rs]: For users that unfamiliar with this filter, I think that it is kind to add a link to the docs.rs page as follows.

```rust
/// Sets [`KVFilter`].
///
/// [`KVFilter`]: https://docs.rs/slog-kvfilter/0.6/slog_kvfilter/struct.KVFilter.html
```

File: src/terminal.rs [hunk 2]
```diff
@@ -70,23 +76,52 @@ impl TerminalLoggerBuilder {
...
+        fn finalfuse<D>(pars: &TerminalLoggerBuilder, drain: D) -> Logger
+            where
+                D: Drain + Send + 'static,
+                D::Err: Debug {
+            let drain = Async::new(drain.fuse())
+                .chan_size(pars.channel_size)
+                .build()
+                .fuse();
+            let drain =
+                AtomicSwitch::new(pars.level.set_level_filter(drain).fuse());
```
Comment (sile) [inline src/terminal.rs:108]: It seems that there is no way to obtain `AtomicSwitchCtrl` from the drain after creating the logger (note I have never used `AtomicSwitch`).

Comment (przygienda) [reply src/terminal.rs:108]: I checked, there is ...

File: src/file.rs (hunk 1 above)
Comment (przygienda) [reply src/file.rs]: sure ,you want me to add it ?

Comment (sile) [reply src/file.rs]: Yes, please. > you want me to add it ?

File: src/terminal.rs (hunk 2 above)
Comment (sile) [reply src/terminal.rs:108]: It seems helpful to add an example code (rustdoc documentation) that explains the existence and an usage of `AtomicSwitch`.

File: src/file.rs (hunk 1 above)
Comment (przygienda) [reply src/file.rs]: Done

File: src/terminal.rs (hunk 2 above)
Comment (przygienda) [reply src/terminal.rs:108]: Yupp, you were correct. The way I've hidden the Atomic it's useless. I have to refactor that which is more complex than it looks due to Drain having so many requirements as trait to write generics over it. I have to solve the case and will get back once I have that done.

**Comment by przygienda** [comment]
Needs new pull once Atomic is refactored ...

**Pull request closed without merging**